modules/darwinosl/darwinosl-source.m

```diff
@@ -0,0 +1,559 @@
+                                                              g_str_hash(self->options.filter_predicate) :
+                                                              0);
+  return TRUE;
+}
+
+static void
+_close_osl(LogThreadedSourceDriver *s)
+{
+  DarwinOSLogSourceDriver *self = (DarwinOSLogSourceDriver *)s;
+  [self->osLogSource closeAll];
+}
+
+/* runs in a dedicated thread */
+static LogThreadedFetchResult
+_fetch(LogThreadedSourceDriver *s)
+{
+  DarwinOSLogSourceDriver *self = (DarwinOSLogSourceDriver *)s;
+
+  gboolean fetchedEnough = (self->options.fetch_limit && self->curr_fetch_in_run > self->options.fetch_limit);
+  OSLogEntry *nextLogEntry = (fetchedEnough ? nil :[self->osLogSource fetchNextEntry]);
+
+  if (nextLogEntry == nil)
+    {
+      msg_trace(nextLogEntry ? "darwinosl: Fetch limit reached" : "darwinosl: No more log data currently");
+      msg_debug("darwinosl: Fetched logs in this run", evt_tag_long("fetch count", (long) self->curr_fetch_in_run));
+      LogThreadedFetchResult result = {THREADED_FETCH_NO_DATA, NULL};
+      return result;
+    }
+
+  self->curr_fetch_in_run++;
+
+  const char *log_string = [self->osLogSource stringFromDarwinOSLogEntry:nextLogEntry].UTF8String;
+  msg_trace("darwinosl: Msg string is composed", evt_tag_str("msg", log_string));
+
+  /* NOTE: No other options unfortunately (there is no "current position" like getter)
+            If there are multiple occurrences of the same time, the method returns the earliest occurrence.
+            So, in a worst case there might be log duplications on restart from bookmark
+  */
+  self->log_source_position.log_position = [nextLogEntry.date timeIntervalSince1970];
+  self->log_source_position.last_msg_hash = g_str_hash(log_string);
+
+  LogSource *worker = (LogSource *) self->super.worker;
+  Bookmark *bookmark = ack_tracker_request_bookmark(worker->ack_tracker);
+  darwinosl_source_persist_fill_bookmark(self->log_source_persist, bookmark, self->log_source_position);
+  msg_trace("darwinosl: Bookmark created",
+            evt_tag_printf("position", "%.6f", self->log_source_position.log_position),
+            evt_tag_printf("last message hash", "%u", self->log_source_position.last_msg_hash),
+            evt_tag_printf("filter predicate hash", "%u", self->log_source_position.last_used_filter_predicate_hash));
+
+  LogMessage *msg;
+  gsize msg_len = _log_message_from_string(log_string, self->options.format_options, &msg);
+  LogThreadedFetchResult result = {THREADED_FETCH_SUCCESS, msg};
+  _log_reader_insert_msg_length_stats(self, msg_len);
+  return result;
+}
+
+static void
+_send(LogThreadedSourceDriver *s, LogMessage *msg)
+{
+  log_threaded_source_blocking_post(s, msg);
+}
+
+static void
+_request_exit(LogThreadedSourceDriver *s)
+{
+  DarwinOSLogSourceDriver *self = (DarwinOSLogSourceDriver *) s;
+  g_atomic_counter_set(&self->exit_requested, TRUE);
+}
+
+/* TODO: Using some iv_timer or similar instead ?! */
```

Comment:
this is a threaded source, so we don't need to use ivykis, a blocking call is fine.